api/views.py

```diff
@@ -17,6 +18,56 @@ class LogViewSet(viewsets.ModelViewSet):
     queryset = Log.objects.all()
     serializer_class = LogSerializer
 
+    LIST_CACHE_KEY = 'log_list'
+    DETAIL_CACHE_KEY_PREFIX = 'log_details_'
+
+    def _invalidate_list_cache(self):
+        cache_key = self.LIST_CACHE_KEY
+        cache.delete(cache_key)
+
+    def list(self, request):
+        cache_key = self.LIST_CACHE_KEY
+
+        data = cache.get(cache_key)
+        if data:
+            return Response(data)
+
+        response = super().list(request)
+        cache.set(cache_key, response.data)
+        return response
+
+    def retrieve(self, request, pk=None):
+        cache_key = f'{self.DETAIL_CACHE_KEY_PREFIX}{pk}'
+
+        data = cache.get(cache_key)
+        if data:
+            return Response(data)
+
+        response = super().retrieve(request, pk=pk)
+        cache.set(cache_key, response.data)
+        return response
+
+    def create(self, request):
+        response = super().create(request)
+        self._invalidate_list_cache()
+        return response
+
+    def update(self, request, pk=None):
+        cache_key = f'{self.DETAIL_CACHE_KEY_PREFIX}{pk}'
+
+        response = super().update(request, pk=pk)
+        cache.set(cache_key, response.data)
```

Comment:
I believe this line won't have any impact? as we are invalidating cache on the next line.

Reply:
also, we should invalidate the detail cache.

Reply:
Line 59 updates the detail cache and next line invalidates the list cache